### ISSUE TITLE
need to cherry pick fix for istioctl fails with Error: an empty namespace may not be set during creation

### ISSUE THREAD

**Issue by andraxylia** [istio/istio#1282]
The istioctl command fails with the latest kubernetes version > 1.7.6 , because, as opposed to kubectl, it does not populate the namespace with the "default" namespace:

istioctl create -f samples/bookinfo/kube/route-rule-all-v1.yaml 

This has to be cherry-picked into 0.2 release.

Lomg-term, istioctl should use the same go library as kubectl, to avoid duplicating work.

**Comment by andraxylia** [comment]
cc @xiaolanz

**Comment by ldemailly** [comment]
>= ? when exactly did this change/break ? is it sure 1.7.x changed that behavior, in which case it sounds like a bug that k8s have changed that in a patch release?

**Comment by andraxylia** [comment]
1.7.6

It's not a bug in k8s, it'a an extra check.

**Comment by ldemailly** [comment]
this seems to be a dup of 
https://github.com/istio/pilot/issues/1472
and
https://github.com/istio/pilot/issues/1637

which @ayj fixed in https://github.com/istio/pilot/pull/1499 but that didn't make it to release-0.2 ? (or is not fully fixed?) - though I see another commit more recent https://github.com/istio/pilot/pull/1638

**Comment by ldemailly** [comment]
I can confirm that istioctl on master works fine:
```
 ~/go/bin/istioctl create -f /tmp/test.rule 
Created config route-rule/default/ratings-default at revision 2901197
```

**Comment by xiaolanz** [comment]
Andra, I assigned it back to you since it's fixed. Feel free to assign bugs to me if you need my help.

**Comment by andraxylia** [comment]
Great , thanks @xiaolanz 
Closing this issue, since it was fixed in istio/pilot#1638. We just need a cherry-pick into 0.2.
